apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php

```diff
@@ -25,17 +25,42 @@ class Full_Site_Editing {
 	 */
 	private $template_post_types = [ 'wp_template' ];
 
+	/**
+	 * Current theme slug.
+	 *
+	 * @var string
+	 */
+	private $theme_slug = '';
+
+	/**
+	 * Instance of WP_Template_Inserter class.
+	 *
+	 * @var WP_Template_Inserter
+	 */
+	public $wp_template_inserter;
+
+	/**
+	 * List of FSE supported themes.
+	 *
+	 * @var array
+	 */
+	const SUPPORTED_THEMES = [ 'modern-business' ];
+
 	/**
 	 * Full_Site_Editing constructor.
 	 */
 	private function __construct() {
-		add_action( 'init', array( $this, 'register_blocks' ), 100 );
-		add_action( 'init', array( $this, 'register_template_post_types' ) );
-		add_action( 'enqueue_block_editor_assets', array( $this, 'enqueue_script_and_style' ), 100 );
-		add_action( 'the_post', array( $this, 'merge_template_and_post' ) );
-		add_filter( 'wp_insert_post_data', array( $this, 'remove_template_components' ), 10, 2 );
-		add_filter( 'admin_body_class', array( $this, 'toggle_editor_post_title_visibility' ) );
-		add_filter( 'block_editor_settings', array( $this, 'set_block_template' ) );
+		add_action( 'init', [ $this, 'register_blocks' ], 100 );
+		add_action( 'init', [ $this, 'register_template_post_types' ] );
+		add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_script_and_style' ], 100 );
+		add_action( 'the_post', [ $this, 'merge_template_and_post' ] );
+		add_filter( 'wp_insert_post_data', [ $this, 'remove_template_components' ], 10, 2 );
+		add_filter( 'admin_body_class', [ $this, 'toggle_editor_post_title_visibility' ] );
+		add_filter( 'block_editor_settings', [ $this, 'set_block_template' ] );
+		add_action( 'after_switch_theme', [ $this, 'insert_theme_template_data' ] );
```

Comment:
In theory this could be switched from `after_switch_theme` to `init` too. That way we would ensure that template date is populated in all environments. It shouldn't impact performance because we have early bails for supported theme and template insertion option indicator (cached).

Interested to hear your thoughts.

Reply:
After some thought - one benefit of tying it to `init` is that we can rollout emergency template updates for all sites by just bumping the option version. 🤔 That's not really possible with theme switch as far as I can see - we'd have to write separate scripts to handle it, and it would only work on platforms that we control.

Reply:
After some more though 😄 
> one benefit of tying it to init is that we can rollout emergency template updates for all sites by just bumping the option version.
This won't work properly without some special handling, because we'd also overwrite templates that were modified (non-default ones).

Reply:
I think this hook feels appropriate for now.

If we need to roll out template updates, there's still pathways available with async jobs and migration scripts that run transforms on the related post_content. We can also re-evaluate if this become a more specific issue.